test/parallel/test-event-emitter-max-listeners.js

```diff
@@ -31,12 +30,25 @@ e.on('maxListeners', common.mustCall());
 e.setMaxListeners(42);
 
 const throwsObjs = [NaN, -1, 'and even this'];
-const maxError = /^TypeError: "n" argument must be a positive number$/;
-const defError = /^TypeError: "defaultMaxListeners" must be a positive number$/;
 
 for (const obj of throwsObjs) {
-  assert.throws(() => e.setMaxListeners(obj), maxError);
-  assert.throws(() => events.defaultMaxListeners = obj, defError);
+  common.expectsError(
+    () => e.setMaxListeners(obj),
+    {
+      code: 'ERR_INVALID_ARG_TYPE',
+      type: TypeError,
+      message: 'The "n" argument must be of type positive number'
```

Comment:
This is a really confusing error message. I think we have a more appropriate error message for things like that.